drivers/api/no_os_dma.c

```diff
@@ -303,6 +303,7 @@ int no_os_dma_config_xfer(struct no_os_dma_desc *desc,
 		no_os_irq_set_priority(desc->irq_ctrl, ch->irq_num, xfer[0].irq_priority);
 	}
 
+	no_os_mutex_unlock(ch->mutex);
```

Comment:
this isn't enough. you need to call it even if an error occurs.

Reply:
yes, and we need to clear the list too.

Reply:
But the unlock is allready called on error right? check line 311 below:

Reply:
i modified the patch and added code to empty the list we filled in case of error.
so i think we need the error path handling with this.